### ISSUE TITLE
Utrecht Ms. 406 IIIF Manifest

### ISSUE THREAD

**Issue by napulen** [DDMAL/cantus#319]
When importing the Utrecht Ms. 406 manuscript, its IIIF manifest seems to be breaking the logic of our frontend.

I did an initial inspection to this manifest and its structure looks similar to every other one we have used (and it should work), however, something particular is that many of its links end with trailing slashes rather than filenames and extensions:

```
Usually
https://www.diamm.ac.uk/sources/1355/manifest/manifest.json
etc...

This one
https://www.diamm.ac.uk/sources/1355/manifest/
etc...
```

We need to check whether that is the source of the problem

**Comment by ahankinson** [comment]
You can't really assume that all IIIF manifests will be found at consistent URLs, and `manifest.json` isn't required. The DIAMM pattern is recommended by the IIIF spec, but (for example), Bodleian IIIF manifests look like this:

http://iiif.bodleian.ox.ac.uk/iiif/manifest/819148d3-c292-4ba8-ab1f-52f696ee10fa.json

**Comment by napulen** [comment]
I don't think this has been an intentional assumption by anyone. This is merely something I've observed looking at the file and might as well be unrelated to the issue.

I'll know for sure once I turn the debugger whether this is the issue.

Thanks for letting me know about the recommended pattern of the IIIF spec. Taking that into account.

**Comment by eliwaksbaum** [comment]
I can't even look at this one. All i see is `detail: "Authentication credentials were not provided."`

**Comment by annamorphism** [comment]
@eliwaksbaum the images on DIAMM are also hidden unless the user logs in; I can get at the manifest using the link when I am already logged into DIAMM. 
The images are viewable without any kind of log-in directly from the UU site; maybe it's possible to get at the manifest from there too? (not obvious how...) https://objects.library.uu.nl/reader/index.php?obj=1874-327915

**Comment by dchiller** [comment]
- I think that the Utrecht library uses an IIP server -- if you go to the advanced zoom view for the manuscript on their site (https://objects.library.uu.nl/reader/viewer.php?obj=1874-327915&pagenum=1&lan=en) you'll see that they are making requests for the various sections of the image (just as we do) from iipsrv. As I've looked at the Diva code that we use, it seems that it does support IIP images? But that would be a pretty major depature...
- Especially since DIAMM does have an IIIF manifest of images ~ do you think they would let us have a service account?

**Comment by annamorphism** [comment]
tagging @fujinaga since this seems to have turned into a question about consulting with one of our partner projects...

**Comment by fujinaga** [comment]
I don't understand the issue with MS. 406. Why can we not put this on Cantus Ultimus?

**Comment by dchiller** [comment]
The issue is that we have two sources of images. One source, the library at the University Utrecht, does not seem to serve via IIIF. The other source is DIAMM, which does serve images via IIIF, but requires log-in with a free account to access those images. 

When I make a requests to the manifest url from DIAMM (https://www.diamm.ac.uk/sources/1355/manifest/), I get a 403 error with the message "Authentication credentials were not provided". If I log in to DIAMM and then make the request in the browser in that same session, I can access the manifest. 

So, it seems like we need authentication credentials that we can provide programmatically to access the IIIF manifest.

**Comment by dchiller** [comment]
Reviewing issues for Cantus Ultimus and wanting to turn this issue into something actionable. @fujinaga @annamorphism, I think this is a decision that I need your help on. 

I think the options are as follows:
1. Do not support Utrecht Ms 406 on Cantus Ultimus.
2. Support IIP image servers with Diva.js (this was previously deprecated in Diva, it seems -- eg. **https://github.com/DDMAL/diva.js/commit/05e291f064b1fd35c4b704af3e26cfb9f626da32**). Re-supporting IIP would allow us to serve the images from the [Universiteitsbibliotheek](https://www.uu.nl/universiteitsbibliotheek).
3. Obtain service credentials from DIAMM and update Cantus Ultimus to support authentication for IIIF manifests. This way, we could serve images for Utrecht Ms 406 from DIAMM's IIIF server.

Thoughts?

**Comment by ahankinson** [comment]
We can talk about this when I'm in Montreal.

**Comment by dchiller** [comment]
Ah, ok!

Yes, let's put this on the list for discussion! -- Looking forward to it.

**Comment by fujinaga** [comment]
Actually, we do have a Discussion section for CU:
e.g.: https://github.com/DDMAL/cantus/discussions/383
So can you (@dchiller)move this there?

**Comment by dchiller** [comment]
Perfect! Moved to #705.

**Comment by dchiller** [comment]
In responding to a question from Debra, I discovered that at some point in the last few months, the Utrecht library has added an IIIF manifest to the record. So we could potentially serve images from there!
